resources/testharness.js

```diff
@@ -1706,6 +1706,62 @@
     }
     expose_assert(assert_array_approx_equals, "assert_array_approx_equals");
 
+    /**
```

Comment:
maybe for ease of maintenance, we should keep these assertions in the `webnn` directory - I don't expect many other test suites will need these assertions.

Reply:
Thanks @dontcallmedom. Yes, I agree with you, I'll update this PR.

Reply:
@dontcallmedom I split out [#35789 adding ULP based assertion](https://github.com/web-platform-tests/wpt/pull/35789) from this PR for better review, PTAL, thanks.

Then I will update this PR based on [#35789](https://github.com/web-platform-tests/wpt/pull/35789).